### PR TITLE
Update `PFMERGE` to be supported as multishard command

### PR DESCRIPTION
**Pull request by lucifercr07**
- Updated `PFMERGE` command to support as multishard command
- Added `preprocess`, `decompose`, `compose` implementation for `PFMERGE`
- Added unit tests
- Update integration tests (Disabled HTTP/WS tests as multishard command not supported for those)

image

### CODE REVIEW

File: internal/iothread/cmd_compose.go [hunk 1]
```diff
@@ -262,3 +262,17 @@ func composeFlushDB(responses ...ops.StoreResponse) interface{} {
 
 	return clientio.OK
 }
+
+// composeMSet processes responses from multiple shards for an "MSet" operation
+// (Multi-set operation). It loops through the responses to check if any shard returned an error.
+// If an error is detected, it immediately returns that error. Otherwise, it returns "OK"
+// to indicate that all "MSet" operations across shards were successful.
```
Comment (apoorvyadav1111) [inline internal/iothread/cmd_compose.go:269]: Just want to clarify if this note is for PFMERGE here?

**Review by apoorvyadav1111 — APPROVED** [review]
Approved, the changes look good to me. I have left a nit comment about a function doc. 

Thanks,
Apoorv

File: internal/iothread/cmd_compose.go (hunk 1 above)
Comment (apoorvyadav1111) [reply internal/iothread/cmd_compose.go:269]: Please go ahead with the merge if this is exactly as it is supposed to be.

Comment (lucifercr07) [reply internal/iothread/cmd_compose.go:269]: Apologies copy/paste mistake, will fix this.

File: internal/iothread/cmd_preprocess.go [hunk 2]
```diff
@@ -79,3 +79,32 @@ func customProcessCopy(thread *BaseIOThread, diceDBCmd *cmd.DiceDBCmd) error {
...
+
+// preProcessPFMerge prepares the PFMERGE command for preprocessing by sending GET commands
+// to retrieve the value of all the keys to be merged with. The retrieved value is used later in the
+// decomposePFMERGE function to merge the hll keys to the new key.
+func preProcessPFMerge(thread *BaseIOThread, diceDBCmd *cmd.DiceDBCmd) error {
+	if len(diceDBCmd.Args) < 1 {
+		return diceerrors.ErrWrongArgumentCount("PFMERGE")
+	}
+
+	for i := 1; i < len(diceDBCmd.Args); i++ {
```
Comment (AshwinKul28) [inline internal/iothread/cmd_preprocess.go:91]: Please add a comment about explicit call to the internal GETOBJECT command

**Review by AshwinKul28 — APPROVED** [review]
Amazing work. Thanks for this @lucifercr07 . Looks good to me. Just left one comment. Otherwise good to merge.

File: internal/iothread/cmd_preprocess.go (hunk 2 above)
Comment (lucifercr07) [reply internal/iothread/cmd_preprocess.go:93]: Thanks, added comment for the same.

**Pull request merged**